include/sparrow/array_api.hpp

```diff
@@ -33,28 +43,96 @@ namespace sparrow
+         * of the layout is transferred to the array.
+         *
+         * @param a An rvalue refernece to the typed layout.
+         */
         template <layout A>
         requires (not std::is_lvalue_reference_v<A>)
-        array(A&& a);
-
+        explicit array(A&& a);
+
+        /**
+         * Constructs an array from the given typed layout. The ownership of
+         * the layout is not transferred to the array and the layout's lifetime
+         * must be longer than that of the array.
+         *
+         * @param a A pointer to the typed layout.
+         */
         template <layout A>
-        array(A* a);
-
+        explicit array(A* a);
+
+        /**
+         * Constructs an array from the given typed layout. The ownership of
+         * the layout is shared by this array and any other shared pointer
+         * referencing it.
+         *
+         * @param a A shared pointer holding the layout.
+         */
         template <layout A>
-        array(std::shared_ptr<A> a);
-
+        explicit array(std::shared_ptr<A> a);
+
+        /**
+         * Constructs an array from the given Arrow C structures, whose ownerhips
+         * is tranferred to the array. The user should not use \c array nor \c schema
+         * after calling this constructor.
+         *
+         * @param array The ArrowArray structure to transfer into the array.
+         * @param schema The ArowSchema structure to transfer into the array.
+         */
         SPARROW_API array(ArrowArray&& array, ArrowSchema&& schema);
+
+        /**
+         * Constructs an array from the given Arrow C structures. The array takes
+         * the ownerhship of the ArrowArray only. The used should not use \c array
+         * after calling this constructor. \c shcema can still be used normally.
+         *
+         * @param array The ArrowArray structure to transfer into the array.
+         * @param schema The ArrowSchema to reference in the array.
+         */
         SPARROW_API array(ArrowArray&& array, ArrowSchema* schema);
+
+        /**
+         * Constructs an array from the given Arrow C structures. Both structures
+         * are referenced from the array and can still be used normally after calling
+         * this cnstructor.
+         *
+         * @param array The arrowArray structure to reference in the array.
+         * @param schema The ArrowSchema to reference in the array.
+         */
         SPARROW_API array(ArrowArray* array, ArrowSchema* schema);
         
+        /**
+         * Return the number of elements in the array.
+         */
         SPARROW_API size_type size() const;
+
+        /**
+         * Returns a constant reference to the element at specified \c index.
+         *
+         * @param index The position of the element in the array.
+         * @return A constant reference to the element.
+         */
         SPARROW_API const_reference operator[](size_type) const;
```

Comment:
can you add `index` in the signature
```suggestion
        SPARROW_API const_reference operator[](size_type index) const;
```